drivers/serial/uart_realtek_rts5912.c

```diff
@@ -31,6 +34,76 @@ struct uart_rts5912_device_config {
 struct uart_rts5912_dev_data {
 };
 
+DT_INST_FOREACH_STATUS_OKAY(PINCTRL_DT_INST_DEFINE);
+#if defined(CONFIG_PM)
+#include <reg/reg_gpio.h>
+#include "zephyr/drivers/gpio/gpio_rts5912.h"
+uint32_t rx_wakeup_pin_num;
+uint32_t *rts5912_rx_wake_reg;
+struct gpio_dt_spec uart_rx_wakeup = GPIO_DT_SPEC_GET(DT_NODELABEL(uart0_wrapper), gpios);
+struct k_work_delayable rx_refresh_timeout_work;
+
+static void uart_rts5912_rx_refresh_timeout(struct k_work *work)
+{
+	ARG_UNUSED(work);
+
+	pm_policy_state_lock_put(PM_STATE_SUSPEND_TO_IDLE, PM_ALL_SUBSTATES);
+}
+/* Hook to count entry/exit */
+static void notify_pm_state_entry(enum pm_state state)
+{
+	switch (state) {
+	case PM_STATE_SUSPEND_TO_IDLE:
+		irq_enable(rx_wakeup_pin_num);
+		gpio_rts5912_set_wakeup_pin(rx_wakeup_pin_num);
+		break;
+	default:
+		break;
+	}
+}
+/* Hook to count entry/exit */
+static void notify_pm_state_exit(enum pm_state state)
+{
+	switch (state) {
+	case PM_STATE_SUSPEND_TO_IDLE:
+		uint32_t interrupt_pin = gpio_rts5912_get_intr_pin(rts5912_rx_wake_reg);
+
+		pinctrl_apply_state(PINCTRL_DT_DEV_CONFIG_GET(DT_NODELABEL(uart0_wrapper)),
+				    PINCTRL_STATE_DEFAULT);
+#ifdef CONFIG_UART_CONSOLE_INPUT_EXPIRED
+		if (interrupt_pin == uart_rx_wakeup.pin) {
+			k_timeout_t delay = K_MSEC(CONFIG_UART_CONSOLE_INPUT_EXPIRED_TIMEOUT);
+
+			pm_policy_state_lock_get(PM_STATE_SUSPEND_TO_IDLE, PM_ALL_SUBSTATES);
+			k_work_reschedule(&rx_refresh_timeout_work, delay);
+		}
+#endif
+		NVIC_ClearPendingIRQ(rx_wakeup_pin_num);
+		irq_disable(rx_wakeup_pin_num);
+		break;
+	default:
+		break;
+	}
+}
+static struct pm_notifier notifier = {
+	.state_entry = notify_pm_state_entry,
+	.state_exit = notify_pm_state_exit,
+};
+
+static void uart_rx_wait(const struct device *dev, void *user_data)
+{
+	ARG_UNUSED(user_data);
+#ifdef CONFIG_UART_CONSOLE_INPUT_EXPIRED
```

Comment:
can this and the other be a `if (IS_ENABLED(CONFIG_UART_CONSOLE_INPUT_EXPIRED)) {`?